gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/BaselineJavaVersionsExtension.java

```diff
@@ -74,42 +85,41 @@ public final void setLibraryTarget(String value) {
                     + "(Java 15 preview cannot be run on Java 17, e.g.) it is unsuitable for use on projects that"
                     + " are published as libraries.");
         }
-        libraryTarget.set(version.javaLanguageVersion());
+        getLibraryTarget().set(version.javaLanguageVersion());
     }
 
-    /**
-     * Target {@link ChosenJavaVersion} for compilation of code used within distributions,
-     * but not published externally.
-     */
+    /** @deprecated Use {@link #getDistributionTarget()} instead. */
+    @Deprecated
     public final Property<ChosenJavaVersion> distributionTarget() {
-        return distributionTarget;
+        return getDistributionTarget();
     }
 
     @Override
     public final void setDistributionTarget(int value) {
-        distributionTarget.set(ChosenJavaVersion.of(value));
+        getDistributionTarget().set(ChosenJavaVersion.of(value));
     }
 
     /** Accepts inputs such as '17_PREVIEW'. */
     @Override
     public final void setDistributionTarget(String value) {
```

Comment:
Because these setter overrides exist, Gradle won't let us create an abstract `getDistributionTarget()` property, which is extremely frustrating. We need these to remain as we need to be able to let people set a plain string or integer. Unsure whether this is intended behaviour by Gradle or not.

Reply:
I don't particularly want to call the actual property something else; I guess we just abandon this idea.

Reply:
Gradle bug report: https://github.com/gradle/gradle/issues/26691#issuecomment-2277814640